src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala

```diff
@@ -191,82 +191,92 @@ private[internal] trait TypeConstraints {
     *  @param upper      When `true` search for max solution else min.
     */
   def solve(tvars: List[TypeVar], tparams: List[Symbol], variances: List[Variance], upper: Boolean, depth: Depth): Boolean = {
+    // We're going to be iterating over these a lot. Let's put them in an array.
+    val _tvars     = tvars.toArray
+    val _tparams   = tparams.toArray
+    val _variances = variances.toArray
+    val _len       = _tvars.length
+    // TODO: can we make this reliably inline its HOF argument when it's a function?
```

Comment:
What made you conclude it's not inlined? I did
```
../build/quick/bin/scalac -opt-warnings -opt:l:inline '-opt-inline-from:**' ../src/reflect/scala/reflect/internal/tpe/TypeConstraints.scala -Yopt-log-inline _
```
and see
```
Inline into scala/reflect/internal/tpe/TypeConstraints.solve: inlined scala/reflect/internal/tpe/TypeConstraints.configForeach$1. Before: 65 ins, inlined: 37 ins.
Inline into scala/reflect/internal/tpe/TypeConstraints.solveOne$1: inlined scala/reflect/internal/tpe/TypeConstraints.configForeach$1. Before: 323 ins, inlined: 37 ins.
Inline into scala/reflect/internal/tpe/TypeConstraints.solveOne$1: inlined scala/reflect/internal/tpe/TypeConstraints.configForeach$1. Before: 365 ins, inlined: 37 ins.
Inline into scala/reflect/internal/tpe/TypeConstraints.solveOne$1: inlined scala/reflect/internal/tpe/TypeConstraints.configForeach$1. Before: 408 ins, inlined: 37 ins.
```

Running cfr, I see all the inlined while loops and no more closure allocations.